packages/babel-template/src/formatters.js

```diff
@@ -0,0 +1,88 @@
+// @flow
+
+export type Formatter<T> = {
+  code: string => string,
+  validate: BabelNodeFile => void,
+  unwrap: BabelNodeFile => T,
+};
+
+export const smart: Formatter<*> = {
+  code: str => str,
+  validate: (ast: BabelNodeFile) => {
+    const { program } = ast;
+    if (program.directives && program.directives.length > 0) {
+      throw new Error("Unexpected directives found");
+    }
+  },
+  unwrap: (
+    ast: BabelNodeFile,
+  ): Array<BabelNodeStatement> | BabelNodeStatement => {
+    const { program } = ast;
+    if (program.body.length > 1) {
+      return program.body;
+    } else {
+      return program.body[0];
+    }
+  },
+};
+
+export const statements: Formatter<*> = {
+  code: str => `{\n${str}\n}`,
+  validate: (ast: BabelNodeFile) => {
+    const { program } = ast;
+    if (program.directives && program.directives.length > 0) {
+      throw new Error("Unexpected directives found");
+    }
+  },
+  unwrap: (ast: BabelNodeFile): Array<BabelNodeStatement> => {
+    return ast.program.body;
+  },
+};
+
+export const statement: Formatter<*> = {
+  code: str => str,
+  validate: (ast: BabelNodeFile) => {
+    const { program } = ast;
+    if (program.directives && program.directives.length > 0) {
+      throw new Error("Unexpected directives found");
```

Comment:
Yeah, you'd have to do
```
template.statement`("string");`();
```
if you wanted to get a string expression statement.

It's the same behavior we have right now, it's just that now it actually throws an error instead of silently discarding the string :P

Reply:
I think we should wrap the code in `{ ... }` so that my example is parsed as an expression statement. Otherwise it is inconsistent with `template.statements`, which wraps the input.

e.g.
```js
template.statements`"string";`(); // [ ExpressionStatement {} ]
template.statement`"string";`(); // Error! 💥 
```

Reply:
They both throw at the moment, but I can certainly tweak this logic so strings are always interpreted as statements and not directives, if you think it'll be more ideal. I can see how the name of the functions might make that more surprising.